task/run-opm-command-oci-ta/0.1/run-opm-command-oci-ta.yaml

```diff
@@ -0,0 +1,183 @@
+        (e.g., [ 'alpha', 'render-template', 'basic', 'v4.18/catalog-template.json']).
+      type: array
+    - name: OPM_OUTPUT_PATH
+      description: Relative path for the opm command's output file (e.g. 'v4.18/catalog/example-operator/catalog.json').
+        Relative to the root directory of given source code (Git repository).
+      type: string
+    - name: IDMS_PATH
+      description: Optional, path for ImageDigestMirrorSet file (e.g. '.tekton/images-mirror-set.yaml').
+      type: string
+      default: ""
+  results:
+    - name: SOURCE_ARTIFACT
+      description: The Trusted Artifact URI pointing to the artifact with
+        the application source code with generated file-based catalog from catalog-template.yml.
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+  steps:
+    - name: check-and-skip-if-needed
+      image: registry.access.redhat.com/ubi9/ubi-minimal:latest
+      results:
+        - name: valid_params
+          description: True or False depending if all required parameters were set.
+      env:
+        - name: OPM_OUTPUT_PATH_PARAM
+          value: $(params.OPM_OUTPUT_PATH)
+        - name: SOURCE_ARTIFACT_PARAM
+          value: $(params.SOURCE_ARTIFACT)
+      command: ["/bin/bash", "-c"]
+      args:
+        - |
+          #!/bin/bash
+          # shellcheck source=/dev/null
+          set -e
+
+          if [[ $# -eq 0 || -z "${OPM_OUTPUT_PATH_PARAM}" ]]; then
+            if [[ $# -eq 0 ]]; then
+              echo "Parameter 'OPM_ARGS' is empty. Skipping OPM execution."
+            else
+              echo "Parameter 'OPM_OUTPUT_PATH' is empty. Skipping OPM execution."
+            fi
+
+            echo -n "${SOURCE_ARTIFACT_PARAM}" | tee "$(results.SOURCE_ARTIFACT.path)"
+            echo -n "false" | tee "$(step.results.valid_params.path)"
+            exit 0
+          fi
+
+          echo -n "true" | tee "$(step.results.valid_params.path)"
+        - "bash"
+        - $(params.OPM_ARGS[*])
+    - name: use-trusted-artifact
+      image: quay.io/konflux-ci/build-trusted-artifacts:latest@sha256:635a890e9f0211b4eb5b8c811d6f06b3ed9ff7da952ebad236fbfe52ba36dbf7
+      args:
+        - use
+        - $(params.SOURCE_ARTIFACT)=/var/workdir/source
+      when:
+        - input: "$(steps.check-and-skip-if-needed.results.valid_params)"
+          operator: in
+          values: ["true"]
+    - name: run-opm-command
+      image: registry.redhat.io/openshift4/ose-operator-registry-rhel9:v4.18
+      workingDir: /var/workdir/source
+      when:
+        - input: "$(steps.check-and-skip-if-needed.results.valid_params)"
+          operator: in
+          values: ["true"]
+      env:
+        - name: OPM_OUTPUT_PATH_PARAM
+          value: $(params.OPM_OUTPUT_PATH)
+      securityContext:
+        runAsUser: 0
+      command: ["/bin/bash", "-c"]
+      args:
```

Comment:
Is there a reason you use this structure over the common structure we use elsewhere? i.e.
```bash
    args: ["$(params.IMAGES[*])"]
    script: |
      #!/bin/bash
```
https://github.com/konflux-ci/build-definitions/blob/main/task/build-image-index/0.1/build-image-index.yaml#L89-L91

Reply:
I was able to modify it for `run-opm-command` step, but I am not able to change it for `check-and-skip-if-needed`
That is due to https://github.com/tektoncd/pipeline/issues/8255

And since this task is not mandatory I wanted to skip executing unnecessary steps when no arguments are set.